### PR TITLE
[hotfix/OBMIN-210] creat query for OAuth2 account

### PR DESCRIPTION
**Pull request by Sershok**
set flag "true" if it old user and created account by OAuth2 long time ago

### CODE REVIEW

File: src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -75,6 +75,7 @@ public boolean registerNewUser(UserRegistrationDto userRegistrationDto) {
     @Override
     public User loginUserWithOAuth2(DefaultOidcUser oauth2User) {
         var optionalUser = findByUsernameOrEmail(oauth2User.getEmail());
+        optionalUser.ifPresent(user -> userRepository.setOAuth2LoginToUserByEmail(oauth2User.getEmail()));
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java:78]: You don't use **user** in the body of optional so just rename it to "**ignored**.

The new behaviour in **loginUserWithOAuth2** has to be covered at least with a unit test

**Review by rpkyrych — APPROVED** [review]
A new unit test is required

**Review by rpkyrych — APPROVED** [review]

File: src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
```
Comment (rpkyrych) [inline src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java:25]: You shouldn't use that annotation. It uses the context

File: src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+class UserServiceTest {
+    public static final String NEW_USER_EMAIL = "user@mail.ua";
+    @MockBean
```
Comment (rpkyrych) [inline src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java:28]: You should use Mock annotation instead

File: src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    private UserRepository userRepository;
+    @Captor
+    private ArgumentCaptor<String> oauth2UserArgumentCaptor;
+    @Autowired
```
Comment (rpkyrych) [inline src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java:32]: InjectMocks should be used instead

**Review by rpkyrych — CHANGES_REQUESTED** [review]
The test class shouldn't use the context

**Review by ennnbaev — APPROVED** [review]
Great!

**Review by rpkyrych — APPROVED** [review]

**Review by ImAndVlad — APPROVED** [review]

**Review by IllyaZuew — APPROVED** [review]

**Review by demus-n — APPROVED** [review]

**Pull request merged**